indexing-service/src/main/java/org/apache/druid/indexing/common/task/TaskIdentitiesProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.common.task;
+
+import org.apache.druid.guice.annotations.ExtensionPoint;
+
+import java.util.Map;
+
+/**
+ * The TaskIdentitiesProvider interface helps add metric tags to tasks.
+ * It's meant to make task management, monitoring, and reporting better by providing extra information
+ * about tasks. Both user-defined and system-generated tags are included, making details about task
+ * performance and characteristics clearer in reports and summaries.
+ */
+@ExtensionPoint
+public interface TaskIdentitiesProvider
```

Comment:
I think `TaskInformationProvider` a simpler and cleaner name. Did we consider that?

Reply:
I believe `Information` might be too broad and doesn't capture the specific intent of this ExtensionPoint, which is focused on delivering task identity details for monitoring and management purposes.

Reply:
Design this as an ExtensionPoint, it will be used for areas other than tags, so don't want to limit what it can do in the name of the class.